Improve Mobile Browser Responsiveness

radchukd opened the pull request:
https://gitcoin.co/issue/ScopeLift/umbra-protocol/218/100026261

apbendi reviewed (changes requested):
Hey @radchukd, overall this is looking great! Thanks so much for picking this up and turning it around so quickly. Really happy with the way it's looking :)

Just did a QA run through the app and found a few regressions that seem to have come along with the changes:

### Receive

Withdrawal seems to have been broken. When I click on "Withdraw" I get the following error toast, with the corresponding console error.

<img width="511" alt="Screen Shot 2021-08-16 at 10 52 28 PM" src="https://user-images.githubusercontent.com/1479802/129656504-218c770f-700b-49e1-89e7-58febf9c364c.png">

<img width="1044" alt="Screen Shot 2021-08-16 at 10 52 35 PM" src="https://user-images.githubusercontent.com/1479802/129656515-0d5bf4b1-173e-41c3-8a6e-f03fc2400c52.png">

### Setup

After I select a name, it does not advance to the next screen, and the copy says "click the arrow" which is not visible in mobile mode. One fix would be a copy change: "click the arrow or swipe left". A better solution would be to advance to the next step automatically, but I don't know why we don't do this already. Do you remember @mds1?

<img width="517" alt="Screen Shot 2021-08-16 at 10 46 48 PM" src="https://user-images.githubusercontent.com/1479802/129656688-e7fe5855-0799-4951-9295-ca9358c3a53b.png">

Similarly, after I sign in step 2, it does not advance  me to the next step. This only becomes evident when I tap the sign button again and get the following toast:

<img width="492" alt="Screen Shot 2021-08-16 at 10 46 27 PM" src="https://user-images.githubusercontent.com/1479802/129656778-db8d740e-9022-4742-b16e-329cf7fb82bb.png">

If I swipe to continue at this point, things do work in the next step, so I think we just need to advance to the next step after the signature is collected.

apbendi commented:
A couple other minor pieces of feedback:

I like the menu, but I wonder if we can afford to make the font a *little bit* bigger and give the links a little more padding.

Also, I'm not sure how I feel about the wallet indicator being in the menu, but I also don't have a better idea. Any thoughts @mds1?

<img width="250" alt="Screen Shot 2021-08-16 at 11 01 20 PM" src="https://user-images.githubusercontent.com/1479802/129657122-a8b47887-4549-4fa5-b571-9b87452287c5.png">

mds1 commented:
Like @apbendi said, this is looking great overall, so thanks @radchukd! Below are my replies to Ben, then some additional thoughts from me. Also I did not look at the code yet but will check that out today

------------------------

> Withdrawal seems to have been broken. When I click on "Withdraw" I get the following error toast, with the corresponding console error.

Related to this, there's some errors in the console when I scan for funds

![image](https://user-images.githubusercontent.com/17163988/129766927-83f3e6b8-e35a-475b-979a-25e8bbf4743d.png)


>  A better solution would be to advance to the next step automatically, but I don't know why we don't do this already. Do you remember @mds1?

We do currently do this. It must have broke somehow. I haven't looked at the code changes yet though so not sure offhand how it happened

> Also, I'm not sure how I feel about the wallet indicator being in the menu, but I also don't have a better idea. Any thoughts @mds1?

The alternative would be to hide the "Umbra" text in the header to give more space for the address. I think I'd prefer doing that so it's easy to make sure you are connected to the right wallet

------------------------

A few other UI thoughts:
1. Can we left-align the footer on mobile? Centering looks a little awkward
2. Mobile cards for the receive table look awesome. Two small thoughts:
    1. On mobile we should use the `formatAddress` helper to the address can be on the same line as "From"
    2. The "Received" text has nothing associated with it. I'm guessing "Received" and "From" were each supposed to be rows?

![image](https://user-images.githubusercontent.com/17163988/129767369-c535d819-65a9-496d-84c2-cae9def5a23e.png)

mds1 commented:
Also, in case you didn't get yourself setup to test sending and receiving, the easiest way is just connect your wallet to Rinkeby, go through the Account Setup steps, then send some ETH to yourself and withdraw it back your wallet

mds1 commented on frontend/src/components/WithdrawForm.vue:
```diff
@@ -0,0 +1,107 @@
+<template>
```
Ah a new component! Totally fine pulling things out into this (that component was getting too big anyway), but it seems something isn't being passed right here cause the receive/withdrawal issues mentioned above

mds1 commented on frontend/src/layouts/AddressSettings.vue:
```diff
@@ -0,0 +1,39 @@
+  props: {
+    userDisplayAddress: {
+      type: String,
+      required: false,
```
I think this should be required, but with `null` or `undefined` as acceptable types

radchukd replied:
But, what if the user has not connected their wallet yet?

mds1 replied:
You'll see in `wallet.ts` where `userAddress` is initialized/defined, and it defaults to `undefined` if a user has not connected their wallet yet. So the prop would still be passed, just with a value of undefined

mds1 commented on frontend/src/layouts/HeaderLinks.vue:
```diff
@@ -0,0 +1,23 @@
+
+export default defineComponent({
+  name: 'HeaderLinks',
+
+  props: {},
```
Nit, but we can remove these two lines

mds1 commented:
Just finished looking over the code and it looks really good! Just added a few small comments above

radchukd commented:
Hi, @apbendi @mds1. I am glad to hear that you like it. Thanks for all the feedback and for the advice on setting up Rinkeby.
I have just pushed a commit, which addresses all of the issues you have mentioned.
Please, let me know if it's all good.
Thanks.

mds1 reviewed (approved):
Everything is looking and working great now, thank you @radchukd for the quick fixes! I'll wait for @apbendi to approve also before rebasing + merging

apbendi reviewed (approved):
Great job @radchukd. This looks good to me. @mds1 can you handle the rebase and merge? Eager to get this deployed!

mds1 commented:
Just pushed one small commit here to handle private key wrapping in advanced mode, then merged!

@radchukd Thank you for the great work here 🙌
